tutorials/developer-tutorials/createNewIdentityAttributes.md

```diff
@@ -0,0 +1,185 @@
+
+4. The extension will ask you if you want to name the asset in the generated contract. For the sake of this tutorial, enter `Car` as the asset name.
+
+5. Choose a location to save the project.  Click `Browse`, then click `New Folder`, and name the project what you want e.g. `carContract`.
+
+> __Pro Tip:__ Avoid using spaces when naming the project!
+
+6. Click `Create` and then select the new folder you just created and click `Save`.
+
+7. Finally, select `Add to workspace` from the list of options.
+
+</details>
+
+<details>
+<summary><b>3. Update our smart contract</b></summary>
+
+Now that we've created our very own identities, we can implement attribute-based access control. To do so, follow the steps below:
+
+1. Click the `Explorer` icon in the left sidebar (it will be at the top and will look like a folder) and navigate to the `car-contract.ts` file in the `src` folder under the contract directory.
+
+2. To deny Jack from carrying out the `createCar` transaction, we need to tweak the generated smart contract slightly. We can replace the function with the following: 
+<details open="true">
+<summary> createCar() </summary>
+
+```
+public async createCar(ctx: Context, carId: string, value: string): Promise<void> {
+    const identity: ClientIdentity = ctx.clientIdentity;
+    // Check if the identity has the 'manufacturer' attribute set to 'true'
+    const checkAttr: boolean = identity.assertAttributeValue('manufacturer', 'true');
+    if (checkAttr) {
+        const exists = await this.carExists(ctx, carId);
+        if (exists) {
+            throw new Error(`The car ${carId} already exists`);
+        }
+        const car = new Car();
+        car.value = value;
+        const buffer = Buffer.from(JSON.stringify(car));
+        await ctx.stub.putState(carId, buffer);
+        } else {
+            throw new Error('You must be a manufacturer to carry out this transaction!');
+        }
+}
+```
+>Note: You will need to import `ClientIdentity` by adding the following line at the top of the file (along with the other imports):
+```
+import { ClientIdentity } from 'fabric-shim';
+```
+</details>
+
+***Save*** the changes.
+
+The above simply checks if the identity trying to access the `createCar()` function has the attribute `manufacturer` set to `true`. If it does, allow the transaction to be executed. If the identity does not have the attribute `manufacturer` (or it isn't set to true), do not let the identity carry out the transaction and show an error message.
+
+> Alternatively, you could use the <a href='https://fabric-shim.github.io/release-1.4/fabric-contract-api.Contract.html#beforeTransaction__anchor'> beforeTransaction()</a> method. This method could be used to seperate your business logic from your permissions.
+
+3. To ensure that the above changes are included within our contract, we must `package`, `install` and then `instantiate` the contract.
```

Comment:
I don't think you need this step. By this point people should know that they now need to do this to test out their smart contract. Also this step isn't telling the user to do anything

Reply:
Fixed